### DISCUSSION TITLE
CoastalApp priorities

### DISCUSSION THREAD

**Discussion by saeed-moghimi-noaa** [noaa-ocs-modeling/CoastalApp#151] [Ideas]
CoastalApp priorities:
1. merge branch to `main`
2. incorporate latest version of WaveWatch III into CoastalApp
3. upgrade NUOPC / NEMS drivers
4. include CDAPS into CoastalApp + include HYCOM in ADCIRC

_Originally posted by @zacharyburnettNOAA in https://github.com/noaa-ocs-modeling/CoastalApp/pull/57#issuecomment-985705599_

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 
## Baroclinic cap testing

**Comment by saeed-moghimi-noaa** [comment]
## HPC stack compliance

**Comment by pvelissariou1** [comment]
CoastalApp priorities (updated):

- [ ] Merge branches develop-takis to develop
- [ ] Rename branch develop to main and master master-delete
- [ ] Test CoastalApp with GFortran using the CoastalApp-testsuite
- [ ] Check ADCIRC with GFortran to see if it passes the relevant tests in CoastalApp-testsuite using control file comparisons
- [ ] If above tests fail, we need to check if still we have BC values mismatches again
- [ ] Recheck if ADCIRC produces abnormally high water levels when forced by PaHM (the issue with PaHM has been fixed)
- [ ] Update the ADCIRC compilation in CoastalApp to use cmake  (need to double check with Zach)
- [ ] Update WW3 submodule to use the develop branch upstream instead the forked Andre's scalability branch
- [ ] Test CoastalApp with the updated WW3 (also need to talk to Carsten/Joseph about WW3)
- [ ] Update the WW3 compilation in CoastalApp to use cmake (need to talk to Ali)
- [ ] Add ICE functionality into CoastalApp
- [ ] Incorporate SCHISM functionality 100% into CoastalApp. Need to talk with Carsten/Joseph about code cleanup, modulefiles, libraries, etc
- [ ] Need SCHISM tests for the updated CoastalApp
- [ ] CoastalApp submodule updates should follow a predefined schedule followed by CoastalApp testing
- [ ] Incorporation of FVCOM into CoastalApp (this is going to be a tedious process)
- [ ] Need FVCOM tests for the updated CoastalApp
- [ ] Build scripts are to be updated and enhance their functionality following the CoastalApp updates (as required)
- [ ] Update the NEMS drivers
- [ ] Include CDEPS into CoastalApp (need to talk to Rocky/Ufuk and the team) - this should be first, before the CMEPS functionality
- [ ] Test CDEPS with real data, along with ATMESH/WW3DATA
- [ ] Include CMEPS functionality into CoastalApp
- [ ] Include an atmospheric model in the application
- [ ] Incorporate the NWM functionality into the updated CoastalApp
- [ ] DO WE NEED TO HAVE ESMF AS A THIRDPARTY LIBRARY IN COASTALAPP? (issues missing esmf in TACC/PW and possibly other platforms)
- [ ] Incorporate the hpc-stack functionality
